var/spack/repos/builtin/packages/m4/package.py

```diff
@@ -33,11 +33,19 @@ class M4(AutotoolsPackage, GNUMirrorPackage):
     patch('secure_snprintf.patch', when='@:1.4.18 os=bigsur')
     # https://bugzilla.redhat.com/show_bug.cgi?id=1573342
     patch('https://src.fedoraproject.org/rpms/m4/raw/5d147168d4b93f38a4833f5dd1d650ad88af5a8a/f/m4-1.4.18-glibc-change-work-around.patch', sha256='fc9b61654a3ba1a8d6cd78ce087e7c96366c290bc8d2c299f09828d793b853c8', when='@1.4.18')
+    # doc/m4.texi (Sysval): Avoid stderr noise during test
+    # https://www.mail-archive.com/m4-patches@gnu.org/msg01208.html
+    patch('http://git.savannah.gnu.org/cgit/m4.git/patch/?id=a1354086',
+          sha256='bfdffa7c2eb01021d5849b36972c069693654ad826c1a20b53534009a4ec7a89', when='@1.4.19')
+    patch('http://git.savannah.gnu.org/cgit/m4.git/patch/?id=cd7f4d15',
+          sha256='9dc5fbd0d5cb1037ab1e6d0ecc74a30df218d0a94bdd5a02759a97f62daca573', when='@1.4.19')
 
     variant('sigsegv', default=True,
             description="Build the libsigsegv dependency")
 
     depends_on('libsigsegv', when='+sigsegv')
+    # As configure detects that we patched doc/m4.texi, it wants makeinfo from texinfo
+    depends_on('texinfo', when='@1.4.19')
```

Comment:
Is there any way to avoid this? I understand the issue, but `texinfo` is a bit of an heavy dependency for just `m4`:
```console
$ spack spec texinfo
Input spec
--------------------------------
texinfo

Concretized
--------------------------------
texinfo@6.5%gcc@11.1.0 patches=12f6edb0c6b270b8c8dba2ce17998c580db01182d871ee32b7b6e4129bd1d23a,1732115f651cff98989cb0215d8f64da5e0f7911ebf0c13b064920f088f2ffe1 arch=linux-ubuntu18.04-broadwell
    ^perl@5.34.0%gcc@11.1.0+cpanm+shared+threads arch=linux-ubuntu18.04-broadwell
        ^berkeley-db@18.1.40%gcc@11.1.0+cxx~docs+stl patches=b231fcc4d5cff05e5c3a4814f6a5af0e9a966428dc2176540d2c05aff41de522 arch=linux-ubuntu18.04-broadwell
        ^bzip2@1.0.8%gcc@11.1.0~debug~pic+shared arch=linux-ubuntu18.04-broadwell
            ^diffutils@3.7%gcc@11.1.0 arch=linux-ubuntu18.04-broadwell
                ^libiconv@1.16%gcc@11.1.0 libs=shared,static arch=linux-ubuntu18.04-broadwell
        ^gdbm@1.19%gcc@11.1.0 arch=linux-ubuntu18.04-broadwell
            ^readline@8.1%gcc@11.1.0 arch=linux-ubuntu18.04-broadwell
                ^ncurses@6.2%gcc@11.1.0~symlinks+termlib abi=none arch=linux-ubuntu18.04-broadwell
                    ^pkgconf@1.8.0%gcc@11.1.0 arch=linux-ubuntu18.04-broadwell
        ^zlib@1.2.11%gcc@11.1.0+optimize+pic+shared arch=linux-ubuntu18.04-broadwell
```

Reply:
Correct, especially that early in the dependency chain of building basic packages:

Done it now. First working replacement was to set it's timestamp it back to the old timestamp:
```py
+    @when('@1.4.19')
+    def patch(self):
+        which("touch")('--reference', 'doc/m4.info', 'doc/m4.text')
```
But it's a bit clumsy and it would better be done in pure python instead of blindly using a feature of GNU touch.

So I think this is lightweight enough and sure to work with every system and filesystem:

Replaced texinfo dependency with touching the files which trigger the makefile to
```py
+    @when('@1.4.19')
+    def patch(self):
+        """ skip texinfo of m4.info for patched m4.texi (patched only a test in it) """
+        touch('doc/m4.info')
+        touch('doc/stamp-vti')
```
@alalazo and @scheibelp: Pushed as amended commit, change is ready for review!